oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/util/ElasticIndexUtils.java

```diff
@@ -106,16 +106,25 @@ public static String fieldName(String propertyName) {
     public static String idFromPath(@NotNull String path) {
         byte[] pathBytes = path.getBytes(StandardCharsets.UTF_8);
         if (pathBytes.length > 512) {
-            try {
-                return new String(MessageDigest.getInstance("SHA-256").digest(pathBytes),
-                        StandardCharsets.UTF_8);
-            } catch (NoSuchAlgorithmException e) {
-                throw new IllegalStateException(e);
-            }
+            return sha256Hash(pathBytes);
         }
         return path;
     }
 
+    /**
+     * Computes the SHA-256 hash of the given byte array and returns it as a UTF-8 string.
+     *
+     * @param input the byte array to hash
+     * @return the SHA-256 hash as a string
+     */
+    public static String sha256Hash(byte[] input) {
```

Comment:
I don't think we should re-use this method: it returns a string that contains strange Unicode characters. Instead, we should use hex encoding or so.

Reply:
Let me add a alternative implementation.